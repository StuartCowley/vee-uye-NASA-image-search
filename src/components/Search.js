import React from "react";
import { useState } from "react";
import "../styles/search.css";

const Search = () => {
  return (
    <>
      <form className="search-form">
        <input className="search-input" type="text" placeholder="moon" />
        <button
          className="search-button"
          type="submit"
          data-testid="button-text"
        >
          Go!
        </button>
      </form>
    </>
  );
};

export default Search;
