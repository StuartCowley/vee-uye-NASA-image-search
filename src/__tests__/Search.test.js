import { getByTestId, render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the search button correctly", () => {
    render(<Search />);
    screen.getByText(/Go/);
  });
});
