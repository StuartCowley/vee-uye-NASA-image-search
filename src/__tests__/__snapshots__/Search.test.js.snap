// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      placeholder="moon"
      type="text"
    />
    <button
      class="search-button"
      data-testid="button-text"
      type="submit"
    >
      Go!
    </button>
  </form>
</DocumentFragment>
`;
